decidim-budgets/app/controllers/decidim/budgets/line_items_controller.rb

```diff
@@ -44,11 +44,11 @@ def destroy
       private
 
       def project
-        @project ||= Project.includes(:budget).find_by(id: params[:project_id], decidim_budgets_budget_id: params[:budget_id])
+        @project ||= Project.where(budget:).includes(:budget).find_by(id: params[:project_id])
       end
 
       def budget
-        @budget ||= project.budget
+        @budget ||= Budget.where(component: current_component).find_by(id: params[:budget_id])
```

Comment:
I think we could move the component condition inside the `find_by`:
```suggestion
        @budget ||= Budget.find_by(id: params[:budget_id], component: current_component)
```